R/STRINGExplore.R

```diff
@@ -0,0 +1,136 @@
+
+  #to-->from directional actions
+
+  tfActions <- paste(interactionSet[interactionSet$is_directional == TRUE &
+                                      interactionSet$a_is_acting == FALSE,]$mode, collapse = "; ")
+  if(tfActions == "") {
+    tfActions <- "none found"
+  }
+  toFrom <- paste("actions that", to, "performs on", from, ":", paste(tfActions, collapse = " "))
+
+  interactions <- paste(nondirectional, "<br />" , fromTo, "<br />", toFrom)
+  return(interactions)
+}
+
+#' STRINGExplore
+#'
+#' \code{STRINGExplore} Utilizes the combined score and protein action information
+#'  available on STRING to create an interactive network of high confidence (>0.8) interaction
+#'
+#' @section Details: The STRING edges on their don't provide a good starting
+#'  point for hypothesizing about the reasons for the existence of high-
+#'  confidence interactions. By combining the easy-to-navigate, interactive
+#'  network from the visNetwork package with tooltips of information from
+#'  high-confidence (>0.8) data from STRING protein_actions dataset, STRINGExplore
+#'  aims to ease the process of deciding on future research avenues. The graph layout takes
+#'  into consideration the betweeness centrality of nodes, and so when inputting a gene list
+#'  from a curated system it's highly likely that highly-centrality nodes will impart some
+#'  important functionality to the system.
+#'
+#' @param geneSys (str) A 5 character string representing the name of
+#' the desired curated gene system user would like to build a
+#' high-confidence interaction network for.
+
+#' @return A labeled list("nodes", "edges", "network") containing the visNetwork
+#' object and the edge and node dataframes resulting from the setup of the network.
+#' May be useful for downstream analysis by the end user.
+#'
+#'
+#' @author \href{https://orcid.org/0000-0003-4762-8797}{Gabriela Morgenshtern} (aut)
+#'
+#' @import readr
+#' @import BiocManager
+#' @import biomaRt
+#' @import visNetwork
+#' @importFrom stats setNames
+#' @importFrom magrittr %>%
+#' @import BiocCheck
+#' @import igraph
+#'
+#'
+#' @examples
+#' STRINGExplore("PHALY")
+#'
+#' @export
+
+STRINGExplore <- function(geneSys) {
+  #### 1. Prepare / Load data:####
+  # alternate / future route: add option to create edges from
+  # high correlation scored expression data (Pearson rho >80%) and network thos
+  geneSys <- SyDBgetSysSymbols(fetchData("SysDB"), geneSys)
+  STRINGactions <- fetchData("STRINGactions")
+  STRINGedges <- fetchData("STRINGedges0.8")
+
+  actionSet <- STRINGactions[STRINGactions$protein1 %in% geneSys &
+                               STRINGactions$protein2 %in% geneSys,]
+  sel <- (STRINGedges$protein1 %in% geneSys) & (STRINGedges$protein2 %in% geneSys)
+  geneSysEdges <- STRINGedges[sel, c("protein1", "protein2")]
+
+  #### 2. igraph setup (visNetwork backbone)####
+  sXG <-  graph.data.frame(geneSysEdges, directed = F)
```

Comment:
Referring to the package this function belongs to ("igraph::")

Reply:
used the @import tag approach for the ROxygen header instead